apps/bfd-db-migrator/src/main/java/gov/cms/bfd/migrator/app/HibernateValidator.java

```diff
@@ -0,0 +1,154 @@
+package gov.cms.bfd.migrator.app;
+
+import com.zaxxer.hikari.HikariDataSource;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import javax.persistence.Entity;
+import javax.sql.DataSource;
+import org.hibernate.HibernateException;
+import org.hibernate.SessionFactory;
+import org.hibernate.boot.Metadata;
+import org.hibernate.boot.MetadataSources;
+import org.hibernate.boot.registry.StandardServiceRegistry;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.cfg.Environment;
+import org.hibernate.tool.hbm2ddl.SchemaValidator;
+import org.reflections.Reflections;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Class for containing logic related to Hibernate validation. */
+public class HibernateValidator {
+
+  private final Logger LOGGER = LoggerFactory.getLogger(HibernateValidator.class);
+
+  /**
+   * Set this to <code>true</code> to have Hibernate log a ton of info on the SQL statements being
+   * run and each session's performance. This should eventually be an application setting of some
+   * kind.
+   */
+  private final boolean HIBERNATE_DETAILED_LOGGING = false;
+
+  private final HikariDataSource dataSource;
+
+  private SchemaValidator schemaValidator;
+
+  private Configuration hibernateConfiguration;
+
+  private final List<String> modelPackagesToScan = new ArrayList<>();
+
+  /**
+   * Instantiates a new Hibernate validator.
+   *
+   * @param dataSource the data source to use
+   * @param modelPackagesToScan the model packages to scan
+   */
+  public HibernateValidator(HikariDataSource dataSource, List<String> modelPackagesToScan) {
+    this.dataSource = dataSource;
+    this.modelPackagesToScan.addAll(modelPackagesToScan);
```

Comment:
can't this class just keep a reference to the list?

Reply:
Yeah, originally I had the ability to add more packages to the validator after it was created, but took it out at some point, so this could just be set to a final reference.